common/src/main/java/com/microsoft/identity/common/internal/telemetry/AndroidTelemetryContext.java

```diff
@@ -67,9 +68,14 @@ private void addApplicationInfo(@NonNull final Context context) {
             String packageName = "";
 
             if (applicationInfo != null) {
-                applicationName = applicationInfo.labelRes == 0 && applicationInfo.nonLocalizedLabel != null ?
-                        applicationInfo.nonLocalizedLabel.toString() : context.getString(applicationInfo.labelRes);
                 packageName = applicationInfo.packageName;
+
+                if (applicationInfo.labelRes == 0) {
+                    applicationName = applicationInfo.nonLocalizedLabel == null ? packageName :
```

Comment:
Consider adding some comments describing the difference between label and package name. Also when might the label be null